### DISCUSSION TITLE
erreur 500, rien dans les logs

### DISCUSSION THREAD

**Discussion by west-side-solutions** [brefphp/bref#811] [Support] [answered]
)
Bonjour,
j'ai déployé une application REST via ApiPlatform sous Symfony avec Bref.
En local, ça tourne correctement  mais sur la version déployé je ne comprends pas pourquoi j'ai des erreurs 500 (dues à un timeout je suppose...).
Le problème c'est quand j'épluche les logs je ne trouve rien... 

Avez vous des pistes pour investiguer le problème ?
merci par avance !

![1](https://user-images.githubusercontent.com/74876321/101157449-7090db80-362a-11eb-87c6-7404a917d003.JPG)
![22](https://user-images.githubusercontent.com/74876321/101158344-dfbaff80-362b-11eb-83d7-67d35071c8de.jpg)
![3](https://user-images.githubusercontent.com/74876321/101157469-7981ad00-362a-11eb-827f-f2823a8a9db0.JPG)

**Comment by mnapoli** [comment]
Bonjour,

Un timeout de Lambda c'est signe que le timeout est ailleurs : 

- à la connexion à la BDD
- à la connexion à une API externe
- autre

Je conseille de configurer tous les timeouts pour autre chose que le défaut, qui est souvent 30 ou 60 secondes. En effet ces timeouts par défaut n'ont aucun sens (pour une bonne expérience utilisateur), et ne collent pas à la limite des 30s d'API Gateway.

Cela permettra de transformer le timeout cryptique et faire remonter l'erreur sous-jacente.

Maintenant je soupçonne 2 possibilités : 

- tu essayes de te connecter à une BDD qui est dans un VPC et Lambda n'est pas dans ce VPC ?
- tu est dans un VPC, et tu essayes d'accéder à une API d'internet, voir https://bref.sh/docs/environment/database.html#limitations

**Comment by west-side-solutions** [comment]
Bonjour,
mon RDS est bien dans le même VPC que la Lambda, j'ai d'ailleurs un timeout de 5 sec paramétré sur la connexion à cette BDD.
Par contre je viens de me rendre compte d'une chose étrange. Quand j'attaque le endpoint avec mon navigateur, j'ai pas le souci ...

je soupçonne donc un problème d'entête http, mais qu'est ce qui justifierais un timeout côté Lambda ?
je continue à creuser...

Merci

**Comment by mnapoli** [comment]
Juste pour être clair : sans NAT Gateway dans le VPC, la Lambda ne peut pas appeler une URL/API sur Internet (y compris l'API AWS). Ça aide ?

**Reply by west-side-solutions** [reply]
Oui j'ai bien pris en compte de cette spécificité, le fait que lambda soit vraiment isolée, mais il n'y a vraiment aucun appel externe depuis les scripts. 
Ce qui me pose problème c'est que les logs soient vides, il faut chercher partout et remettre tout bout de code en question 😭

**Comment by west-side-solutions** [comment, marked as answer]
Hello,
finalement c'était bien un problème de timeout sur l'appel d'une api externe ( j'en avais pas connaissance )... Merci d'avoir pris le temps de me répondre ;)
